core/trino-main/src/main/java/io/trino/operator/output/UnnestingPositionsAppender.java

```diff
@@ -94,4 +120,92 @@ private IntArrayList mapPositions(IntArrayList positions, DictionaryBlock block)
         }
         return IntArrayList.wrap(positionArray);
     }
+
+    private static class DictionaryBlockBuilder
+    {
+        private static final int INSTANCE_SIZE = toIntExact(ClassLayout.parseClass(DictionaryBlockBuilder.class).instanceSize());
+        private final int initialEntryCount;
+        private Block dictionary;
+        private int[] dictionaryIds;
+        private int positionCount;
+        private boolean closed;
+
+        public DictionaryBlockBuilder()
+        {
+            this(1024);
```

Comment:
1024 is based on some kind of empirical knowledge?

Reply:
well, it's a nice, round number ;-).
I haven't really tested the alternatives, this can probably be any smallish number, as the value has little impact on performance.